core/chains/evm/gas/universal_estimator.go

```diff
@@ -0,0 +1,412 @@
+		if e != nil {
+			err = e
+			return
+		}
+
+		// If either of these two values are 0 it could be that the network has extremely low priority fees. We should skip the
+		// connectivity check because we're only going to be charged for the base fee, which is mandatory.
+		if (priorityFeeThreshold.Cmp(assets.NewWeiI(0)) > 0) && (bumpedMaxPriorityFeePerGas.Cmp(assets.NewWeiI(0)) > 0) &&
+			bumpedMaxPriorityFeePerGas.Cmp(priorityFeeThreshold) > 0 {
+			return bumped, fmt.Errorf("bumpedMaxPriorityFeePergas: %s is above market's %sth percentile: %s, bumping is halted",
+				bumpedMaxPriorityFeePerGas, strconv.Itoa(ConnectivityPercentile), priorityFeeThreshold)
+		}
+	} else {
+		// If the network doesn't have a mempool then transactions are processed in a FCFS manner and maxPriorityFeePerGas value is irrelevant.
+		// We just need to cap the value at maxPrice in case maxFeePerGas also gets capped.
+		bumpedMaxPriorityFeePerGas = assets.WeiMin(bumpedMaxPriorityFeePerGas, maxPrice)
+	}
+
+	bumpedMaxFeePerGas, err = LimitBumpedFee(originalFee.FeeCap, currentDynamicPrice.FeeCap, bumpedMaxFeePerGas, maxPrice)
+	if err != nil {
+		return bumped, fmt.Errorf("maxFeePerGas error: %s", err.Error())
+	}
+
+	bumpedFee := DynamicFee{FeeCap: bumpedMaxFeePerGas, TipCap: bumpedMaxPriorityFeePerGas}
+	u.logger.Debugw("bumped dynamic fee", "originalFee", originalFee, "bumpedFee", bumpedFee)
+
+	return bumpedFee, nil
+}
+
+// LimitBumpedFee selects the maximum value between the original fee and the bumped attempt. If the result is higher than the max price it gets capped.
+// Geth's implementation has a hard 10% minimum limit for the bumped values, otherwise it rejects the transaction with an error.
+// See: https://github.com/ethereum/go-ethereum/blob/bff330335b94af3643ac2fb809793f77de3069d4/core/tx_list.go#L298
+//
+// Note: for chains that support EIP-1559 but we still choose to send Legacy transactions to them, the limit is still enforcable due to the fact that Legacy transactions
+// are treated the same way as Dynamic transactions under the hood. For chains that don't support EIP-1559 at all, the limit isn't enforcable but a 10% minimum bump percentage
+// makes sense anyway.
+func LimitBumpedFee(originalFee *assets.Wei, currentFee *assets.Wei, bumpedFee *assets.Wei, maxPrice *assets.Wei) (*assets.Wei, error) {
+	if currentFee != nil {
+		bumpedFee = assets.WeiMax(currentFee, bumpedFee)
+	}
+	bumpedFee = assets.WeiMin(bumpedFee, maxPrice)
+
+	// The first check is added for the following edge case:
+	// If originalFee is below 10 wei, then adding the minimum bump percentage won't have any effect on the final value because of rounding down.
+	// Similarly for bumpedFee, it can have the exact same value as the originalFee, even if we bumped, given an originalFee of less than 10 wei
+	// and a small BumpPercent.
+	if bumpedFee.Cmp(originalFee) == 0 ||
+		bumpedFee.Cmp(originalFee.AddPercentage(MinimumBumpPercentage)) < 0 {
```

Comment:
Not sure if I understand this check. Is this even possible if we're restricting `BumpPercent` to be >= to the `MinimumBumpPercentage` on startup?

Reply:
Never mind, just realized that bumpedFee could have been capped so you still have to check it against the uncapped bump to make sure it follows the 10% min